src/Core/Services/SendService.cs

```diff
@@ -0,0 +1,275 @@
+                    }
+                    break;
+                default:
+                    break;
+            }
+
+            return (send, encryptedFileData);
+        }
+
+        public async Task<List<Send>> GetAllAsync()
+        {
+            var userId = await _userService.GetUserIdAsync();
+            var sends = await _storageService.GetAsync<Dictionary<string, SendData>>(SEND_KEY(userId));
+            return sends.Select(kvp => new Send(kvp.Value)).ToList();
+        }
+
+        public async Task<List<SendView>> GetAllDecryptedAsync()
+        {
+            if (_decryptedSendsCache != null)
+            {
+                return _decryptedSendsCache;
+            }
+
+            var hasKey = await _cryptoService.HasKeyAsync();
+            if (!hasKey)
+            {
+                throw new Exception("No Key.");
+            }
+
+            if (_getAllDecryptedTask != null && !_getAllDecryptedTask.IsCompleted && !_getAllDecryptedTask.IsFaulted)
+            {
+                return await _getAllDecryptedTask;
+            }
+
+            async Task<List<SendView>> doTask()
+            {
+                var decSends = new List<SendView>();
+
+                async Task decryptAndAddSendAsync(Send send) => decSends.Add(await send.DecryptAsync());
+                await Task.WhenAll((await GetAllAsync()).Select(s => decryptAndAddSendAsync(s)));
+
+                decSends.OrderBy(s => s, new SendLocaleComparer(_i18nService)).ToList();
+                _decryptedSendsCache = decSends;
+                return _decryptedSendsCache;
+            }
+
+            _getAllDecryptedTask = doTask();
+            return await _getAllDecryptedTask;
+        }
+
+        public async Task<Send> GetAsync(string id)
+        {
+            var userId = await _userService.GetUserIdAsync();
+            var sends = await _storageService.GetAsync<Dictionary<string, SendData>>(SEND_KEY(userId));
+
+            if (sends == null || !sends.ContainsKey(id))
+            {
+                return null;
+            }
+
+            return new Send(sends[id]);
+        }
+
+        public async Task ReplaceAsync(Dictionary<string, SendData> sends)
+        {
+            var userId = await _userService.GetUserIdAsync();
+            await _storageService.SaveAsync(SEND_KEY(userId), sends);
+            _decryptedSendsCache = null;
+        }
+
+        public async Task SaveWithServerAsync(Send send, byte[] encryptedFileData)
+        {
+
+            var request = new SendRequest(send);
+            SendResponse response = null;
+            if (send.Id == null)
+            {
+                if (send.Type == SendType.Text)
+                {
+                    response = await _apiService.PostSendAsync(request);
+                }
+                else if (send.Type == SendType.File)
+                {
+                    var fd = new MultipartFormDataContent($"--BWMobileFormBoundary{DateTime.UtcNow.Ticks}")
```

Comment:
Is this intended to use `UtcNow` where other areas use `Now` because it's going to the server? Again, just not sure, sorry.

Reply:
This parameter sets the boundary in the `Content-Type: multipart/form-data;` header. To my understanding it can be whatever you want and doesn't need to be unique, but I just [copied it from CipherService](https://github.com/bitwarden/mobile/blob/master/src/Core/Services/CipherService.cs#L560-L561)